src/evaluators/SwitchStatement.js

```diff
@@ -99,6 +112,12 @@ function CaseBlockEvaluation(
     return clause.test === null;
   });
 
+  // Abstract interpretation of case blocks is a significantly different process
+  // from regular interpretation, so we fork off early to keep things tidily separated.
+  if (realm.useAbstractInterpretation) {
```

Comment:
If input is a concrete value we still want to use the interpreter, so rather change this condition to: input instanceof AbstractValue.

Reply:
Makes perfect sense, will do. 

If input is a ConcreteValue but the case selectors result in AbstractValues (which I'm assuming is a possibility) I would think we would still want to take the abstract interpreter path. But if such a thing is possible then I can deal with that wrinkle in a subsequent revision.

Reply:
Yes, better leave that as a failure case for now. Going forward, we'd have to have a way to bail out of the interpreter. See DoWhileStatement.js for an example of how this will work.